### PR TITLE
Add producer access mode examples for Java and C++ client.

### PR DESCRIPTION
**Pull request by shibd**
### Preview

image

image

### Related issue
- https://github.com/apache/pulsar/wiki/PIP-68%3A-Exclusive-Producer
- https://github.com/apache/pulsar-client-cpp/issues/95

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by RobertIndie — APPROVED** [review]

File: docs/client-libraries-producers.md [hunk 1]
```diff
@@ -784,4 +784,41 @@ This example shows how to set the `EnforceUnencrypted` encryption policy.
 
   </TabItem>
 </Tabs>
-````
\ No newline at end of file
+````
+
+## Configure Access mode
```
Comment (Anonymitaet) [inline docs/client-libraries-producers.md:789]:
```suggestion
## Configure access mode
```

File: docs/client-libraries-producers.md [hunk 2]
```diff
@@ -784,4 +784,41 @@ This example shows how to set the `EnforceUnencrypted` encryption policy.
...
+
+Access mode allow applications to require exclusive producer access on a topic in order to achieve a "single-writer" situation.
```
Comment (Anonymitaet) [inline docs/client-libraries-producers.md:791]:
```suggestion
[Access mode](concepts-clients.md#access-mode) allows applications to require exclusive producer access on a topic to achieve a "single-writer" situation.
```

File: docs/client-libraries-producers.md [hunk 3]
```diff
@@ -784,4 +784,41 @@ This example shows how to set the `EnforceUnencrypted` encryption policy.
...
+
+More information about access mode can be found [concepts-clients](concepts-clients.md#access-mode).
```
Comment (Anonymitaet) [inline docs/client-libraries-producers.md:793]:
```suggestion

```

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution! 

1. Soft reminder: do not forget the 2nd task commented in https://github.com/apache/pulsar-client-cpp/pull/201#issuecomment-1512408171

2. Can you provide the code PR link (https://github.com/apache/pulsar-client-cpp/pull/201) in the PR description? So that reviewers can have more context on the doc changes.

**Comment by shibd** [comment]
> Soft reminder: do not forget the 2nd task commented in.

Hi, @Anonymitaet The structure of the current document(next) is very different from the previous document(2.11.x, 2.10.x, etc).

I'm not sure where I should add it. And for the CPP client, We don't seem to manage the version of its documentation.

**Comment by Anonymitaet** [comment]
For versioned docs, how about adding the docs as below?

The location might not be perfect but it's the most relevant place.

Java
image

C++
image

File: docs/client-libraries-producers.md [hunk 4]
```diff
@@ -798,6 +798,8 @@ This example shows how to set producer access mode.
   values={[{"label":"Java","value":"Java"},{"label":"C++","value":"C++"}]}>
 <TabItem value="Java">
 
+  This feature is supported after Java client version 2.8.0.
```
Comment (Anonymitaet) [inline docs/client-libraries-producers.md:801]:
```suggestion
::: note

This feature is supported in Java client 2.8.0 or later versions.

:::
```

File: docs/client-libraries-producers.md [hunk 5]
```diff
@@ -809,6 +811,8 @@ This example shows how to set producer access mode.
 
 <TabItem value="C++">
 
+  This feature is supported after CPP client version 3.1.0.
```
Comment (Anonymitaet) [inline docs/client-libraries-producers.md:814]:
```suggestion
::: note

This feature is supported in C++ client 3.1.0 or later versions.

:::
```

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**